server/src/Models/Fields.hs

```diff
@@ -28,24 +26,34 @@ import qualified Data.CAProvinceCodes as CACodes
 import qualified Data.StateCodes as StateCodes
 import qualified Data.Text as T
 import qualified Web.Stripe.Types as Stripe
+import Data.Int
+import qualified Data.Foldable
 
 -- | Cents are used to do any currency-related arithmetic & are represented
 -- as arbitrary-percision numbers.
 newtype Cents =
-    Cents { fromCents :: Natural }
-    deriving newtype (Eq, Ord, Num, ToJSON, FromJSON)
-    deriving (PersistField, PersistFieldSql) via OverflowNatural -- TODO sand-witch: improve this behavior
+    Cents { fromCents :: Int64 }
```

Comment:
I don't like the idea that type allows `Cents` to be negative.
I wonder how realistic it is to hit the overflow on the Postgres side using plain `Word64`?
Probably, this isn't very realistic given that `2^63` is ~ `9*10^18`.

Also, AFAICS, there are `PersistField` and `PersistFieldSql` instances for `Word64`.

Let me know what you think

Reply:
I don't think we may improve security this way: the same scenario when we may reach negative cents we would reach Word64 underflow and this is equally bad.

Reply:
I guess, you're right, `Num` doesn't save us from underflows for non-negative types anyway 😒

Reply:
I can insert asserts into instances and functions: FromJSON, Read, subtractCents, timesQuantity